sdk/storage/Azure.Storage.Files/src/FileClient.cs

```diff
@@ -1682,6 +1682,167 @@ public virtual FileClient WithSnapshot(string shareSnapshot)
+            await this.UploadRangeFromUrlInternal(
+                sourceUri,
+                range,
+                sourceRange,
+                true, // async
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        /// <summary>
+        /// The <see cref="UploadRangeInternal"/> operation writes a range from an Azure File to another Azure file. 
+        /// This API is supported only for version 2019-02-02 and higher.
+        /// </summary>
+        /// <param name="sourceUri">
+        /// Required. Specifies the URL of the source file, up to 2 KB in length.
+        /// If source is an Azure blob or Azure file, it must either be public or must be authenticated via a 
+        /// shared access signature. If the source is public, no authentication is required to perform the operation.
+        /// </param>
+        /// <param name="range">
+        /// Specifies the range of bytes to be written. Both the start and end of the range must be specified.
+        /// </param>
+        /// <param name="sourceRange">
+        /// Specifies the range of bytes to be written from. Both the start and end of the range must be specified.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{FileInfo}"/> describing the
+        /// state of the file.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Response<StorageFileUploadInfo>> UploadRangeFromUrlInternal(
+            Uri sourceUri,
+            HttpRange range,
+            HttpRange sourceRange,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            using (this.Pipeline.BeginLoggingScope(nameof(FileClient)))
+            {
+                this.Pipeline.LogMethodEnter(
+                    nameof(FileClient),
+                    message:
+                    $"{nameof(this.Uri)}: {this.Uri}\n" +
+                    $"{nameof(sourceUri)}: {sourceUri}");
+                try
+                {
+                    var response = await FileRestClient.File.UploadRangeFromURLAsync(
+                        pipeline: Pipeline,
+                        resourceUri: Uri,
+                        range: range.ToString(),
+                        copySource: sourceUri,
+                        contentLength: default,
+                        sourceRange: sourceRange.ToString(),
+                        async: async,
+                        cancellationToken: cancellationToken)
```

Comment:
Do you need an `operationName` here?

Reply:
No, operationName is hardcoded in FileRestClient.File.  See https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files/src/Generated/FileRestClient.cs#L4899

Reply:
Now that you've changed the method name from Url to Uri, I don't think the default value of operationName is going to work with the diagnostic test.  Running them should make it clear one way or the other though.